### ISSUE TITLE
goToDefinition doesn't work properly

### ISSUE THREAD

**Issue by BohdanZinkevych** [mrmckeb/typescript-plugin-css-modules#247]
In **VSCode**, 'Go to Definition' (Ctrl + click on a classname) doesn't work for me; it opens the GenericMarkup.module.less file, and that's it.

tscondig.json
`
    "plugins": [
      {
        "name": "typescript-plugin-css-modules",
        "options": {
          "classnameTransform": "camelCaseOnly",
          "goToDefinition": "enabled",
        }
      }
    ]
`

**Steps to reproduce:**
1. Open any .tsx file that uses a CSS module.
2. Click on a classname imported from the CSS module.
3. Observe that it navigates to the incorrect place in the opened file.

![image](https://github.com/mrmckeb/typescript-plugin-css-modules/assets/116152055/830e07ed-35f9-40e7-9f3a-4156d7503a0c)
![image](https://github.com/mrmckeb/typescript-plugin-css-modules/assets/116152055/8f993b5c-505d-4f54-86ec-c0b572c74024)

**Expected behavior:**
The CSS module file should open at the correct classname's place.

OS: Windows 11
IDE: VSCode

**Comment by dzsodzso63** [comment]
I have the same problem. I upgraded `typescript` to `5.3.3` and `typescript-plugin-css-modules` to `5.0.2` and `goToDefinition` navigates now to the end of the file. It was working before for us
config:
```
{
        "name": "typescript-plugin-css-modules",
        "options": {
          "classnameTransform": "asIs",
          "customMatcher": "\\.css$",
          "dotenvOptions": {},
          "postcssOptions": {},
          "rendererOptions": {},
          "goToDefinition": true
        }
      }
```
we used v4 with ts4 before, without any problem

**Comment by dzsodzso63** [comment]
Adding a minor info here, before in vscode hovering with cmd key underlined the class before, now it does not happen

**Comment by BohdanZinkevych** [comment]
just in case, I use the latest version of **typescript-plugin-css-modules** and **typescript: '4.9.5'**

**Comment by mrmckeb** [comment]
Hi @BohdanZinkevych, `goToDefinition` is a `boolean` value. You can see here that it defaults to `false`. You should be able to get it to work by setting it to `true`:
https://github.com/mrmckeb/typescript-plugin-css-modules?tab=readme-ov-file#options

**Comment by mrmckeb** [comment]
Hi @dzsodzso63, I think you have a different issue.

Can you please raise a new request with a small reproduction? That will help me investigate further.

Note that this is an experimental feature, and even when working, may not always point to the right line.

**Comment by BohdanZinkevych** [comment]
still doesn't work for me with or without goToDefinition: true

**Comment by BohdanZinkevych** [comment]
@mrmckeb yes, it's true, it leads me to the correct file but not to the correct place(line)

**Comment by dzsodzso63** [comment]
@mrmckeb looks like the very same issue to me

**Comment by mrmckeb** [comment]
Hi all, I'm away at the moment, but will try to take a look at this in the next week or so.

This is an experimental feature, and I'm not sure we'll ever be able to make it work perfectly as TypeScript doesn't provide an API for this. But, we can try to make it better.

**Comment by aleksei-berezkin** [comment]
Hi @mrmckeb. Are you still interested in getting a pull request with a fix for this? I think I understand where the problem is: in [this fragment](https://github.com/mrmckeb/typescript-plugin-css-modules/blob/main/src/helpers/createDtsExports.ts#L102) the implementation does not take into consideration `goToDefinition=true` for the default exported object. Of course, it is not possible to support `goToDefinition` for both named and default exports at the same time, but if `namedExport=false`, it is totally possible to make it work for the default export. For example, the generated `.d.ts` could be:

```ts
let _={pageContent = '',


// (multiple newlines)

title = '',

}
export default _
// No named exports
```

This would also fix #262.

If you are interested, I would be happy to create a PR with the fix. Thank you.

**Comment by aleksei-berezkin** [comment]
Workaround: https://github.com/JetBrains/typescript-plugin-css-modules with `"goToDefinition": "default"`
